iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/webservice/v1/controllers/DeploymentsController.java

```diff
@@ -0,0 +1,114 @@
+     * are provided in the inner exception.
+     */
+    public CompletionStage<Result> getDeployment(final String id, boolean includeDeviceStatus) throws
+            ExternalDependencyException, InvalidInputException {
+        if (StringUtils.isEmpty(id)) {
+            throw new InvalidInputException("Must specify deployment id to retrieve");
+        }
+
+        return this.deploymentsService.getAsync(id, includeDeviceStatus)
+                .thenApply(deployment -> ok(toJson(new DeploymentApiModel(deployment))));
+    }
+
+    /**
+     * Deletes the deployment of the given id.
+     *
+     * @return True if the deployment was deleted successfully.
+     * 404 - Http not found is returned if the id provided doesn't belong to a deployment.
+     * 400 - Bad request is returned if an id is not provided
+     * @throws ExternalDependencyException thrown if there is an issue querying the RegistryManager. Details
+     * are provided in the inner exception.
+     */
+    public CompletionStage<Result> deleteAsync(final String id) throws
+            ExternalDependencyException, InvalidInputException {
+        if (StringUtils.isEmpty(id)) {
+            throw new InvalidInputException("Must specify deployment id to retrieve");
+        }
+
+        return this.deploymentsService.deleteAsync(id).thenApply(result -> ok());
+    }
+
+    /**
+     * Creates a new deployment with the values specified in the body as JSON in the form of
+     * {@link DeploymentApiModel}.
+     *
+     * @return A {@link DeploymentApiModel} modeling the newly created configuration.
+     * 400 - Bad request is returned if deviceGroupId, packageId, name, or priority are invalid.
+     * @throws ExternalDependencyException thrown if there is an issue communicating with the hub or in the
+     * creation of the deployment. Details are provided in the inner exception.
+     */
+    public CompletionStage<Result> postAsync() throws ExternalDependencyException, InvalidInputException {
+        final JsonNode json = request().body().asJson();
+        final DeploymentApiModel deployment = fromJson(json, DeploymentApiModel.class);
+
+        if (StringUtils.isEmpty(deployment.getDeviceGroupId())) {
+            throw new InvalidInputException("DeviceGroupId must be provided");
+        }
+
+        if (StringUtils.isEmpty(deployment.getDeviceGroupQuery())) {
+            throw new InvalidInputException("DeviceGroupQuery must be provided");
+        }
+
+        if (StringUtils.isEmpty(deployment.getPackageContent())) {
+            throw new InvalidInputException("PackageContent must be provided");
+        }
+
+        if (StringUtils.isEmpty(deployment.getName())) {
+            throw new InvalidInputException("Name must be provided");
+        }
+
+        if (deployment.getPriority() < 0) {
+            throw new InvalidInputException("Priority must be provided");
```

Comment:
should the error message be like "Priority must be greater than or equal to zero" ?